mifosng-provider/src/main/java/org/mifosplatform/portfolio/group/service/GroupingTypesWritePlatformServiceJpaRepositoryImpl.java

```diff
@@ -185,6 +186,16 @@ private CommandProcessingResult createGroupingType(final JsonCommand command, fi
 
             // pre-save to generate id for use in group hierarchy
             this.groupRepository.save(newGroup);
+            
+			/* Generate Hierarchy for groups associated to center */
+			if (newGroup.isCenter() && newGroup.hasActiveGroups()) {
+				for (final Group group : groupMembers) {
```

Comment:
Here, group is already a domain object, what is being done in this block of code is to fetch the id from a domain object and then  get the domain object based on the ID. So the objects groupForUpdate and group are the same and we have an unnecessary call to the database

group.generateHierarchyToAssociateGroups(newGroup) would have sufficed